stroom-explorer/stroom-explorer-impl-db/src/main/java/stroom/explorer/impl/db/ExplorerFavDaoImpl.java

```diff
@@ -9,26 +9,26 @@
 import javax.inject.Inject;
 
 import static stroom.explorer.impl.db.jooq.Tables.EXPLORER_FAVOURITE;
+import static stroom.explorer.impl.db.jooq.Tables.EXPLORER_NODE;
 
 public class ExplorerFavDaoImpl implements ExplorerFavDao {
 
-    private final ExplorerDbConnProvider docStoreDbConnProvider;
+    private final ExplorerDbConnProvider explorerDbConnProvider;
 
     @Inject
-    ExplorerFavDaoImpl(final ExplorerDbConnProvider docStoreDbConnProvider) {
-        this.docStoreDbConnProvider = docStoreDbConnProvider;
+    ExplorerFavDaoImpl(final ExplorerDbConnProvider explorerDbConnProvider) {
+        this.explorerDbConnProvider = explorerDbConnProvider;
     }
 
     @Override
     public void createFavouriteForUser(final DocRef docRef, final String userId) {
-        final int count = JooqUtil.contextResult(docStoreDbConnProvider, context -> context
+        final int explorerNodeId = getExplorerNodeId(docRef);
+        final int count = JooqUtil.contextResult(explorerDbConnProvider, context -> context
                 .insertInto(EXPLORER_FAVOURITE,
-                        EXPLORER_FAVOURITE.DOC_TYPE,
-                        EXPLORER_FAVOURITE.DOC_UUID,
+                        EXPLORER_FAVOURITE.EXPLORER_NODE_ID,
                         EXPLORER_FAVOURITE.USER_UUID,
                         EXPLORER_FAVOURITE.CREATE_TIME_MS)
-                .values(docRef.getType(),
-                        docRef.getUuid(),
+                .values(explorerNodeId,
                         userId,
                         Instant.now().toEpochMilli())
                 .onDuplicateKeyIgnore()
```

Comment:
onDuplicateKeyIgnore is dangerous as it hides other errors.  Use JooqUtil.tryCreate() with the explorer_node_id and user_uuid fields as keyField(1|2).